### PR TITLE
gce: when using only ipv6, we don't need a CNI to set up the network.

### PR DESCRIPTION
**Pull request by justinsb**
IPv6 addresses are "real" addresses, and don't need any further
configuration on GCE.

### CODE REVIEW

File: cmd/kops-controller/controllers/gceipam.go [hunk 1]
```diff
@@ -139,6 +147,46 @@ func (r *GCEIPAMReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
...
+			}
+		}
+
+		if allIPv6 {
+			// IPv6 does not require a route to be set up, so mark the node as NetworkReady
+			for _, condition := range node.Status.Conditions {
+				if condition.Type == "NetworkUnavailable" {
+					if condition.Status == corev1.ConditionTrue {
+						newCondition := metav1.Condition{
+							Message: "Node has IPv6",
```
Comment (johngmyers) [inline cmd/kops-controller/controllers/gceipam.go:176]:
```suggestion
							Message: "Node is IPv6-only",
```

**Comment by johngmyers** [comment]
Unrelated to this PR, but consider updating the GCP node identifier to use the non-legacy interface.

File: cmd/kops-controller/controllers/gceipam.go [hunk 2]
```diff
@@ -139,6 +147,46 @@ func (r *GCEIPAMReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		}
 	}
 
+	if len(node.Spec.PodCIDRs) != 0 {
```
Comment (aojea) [inline cmd/kops-controller/controllers/gceipam.go:150]: you can make this simpler , this field is already validated in the apiserver https://github.com/kubernetes/kubernetes/blob/54d42e6a658f84ea66593d3107f41db6931a5fbf/pkg/apis/core/validation/validation.go#L4247-L4281

```suggestion
       netutils "k8s.io/utils/net" 


	if len(node.Spec.PodCIDRs) != 1 {
	    return ctrl.Result{}, nil
	 }
	 if netutils.IsIPv4CIDR(node.Spec.PodCIDRs[0]) {
	      return ctrl.Result{}, nil
	 }
	 ...
	 patch condition here to unset NetworkUnavailable
	 ...
	    
```

**Comment by aojea** [comment]
> /hold as @aojea suggested this might be done for us in the latest CCM

/hold cancel

the cloud provider gcp will take some time because the ipam code changes are not straightforward, this ipam here is pretty neat and simple and will allow us to iterate faster

**Pull request closed without merging**